src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,661 @@
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigSafetyFactor(void* arkode_mem, sunrealtype dom_eig_sfty)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (dom_eig_sfty < SUN_RCONST(1.0))
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "dom_eig_sfty must be greater than or equal to 1");
+    return (ARK_ILL_INPUT);
+  }
+
+  step_mem->dom_eig_sfty = dom_eig_sfty;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetSSPStageNum sets the number of stages in the following
+  SSP methods:
+
+      ARKODE_LSRK_SSP_S_2  -- num_of_stages must be greater than or equal to 2
+      ARKODE_LSRK_SSP_S_3  -- num_of_stages must be a perfect square greater than or equal to 9
+      ARKODE_LSRK_SSP_10_4 -- num_of_stages must be equal to 10 - no need to call!
+
+  This set routine must be called after calling LSRKStepSetMethod with an SSP method
+  ---------------------------------------------------------------*/
+int LSRKStepSetSSPStageNum(void* arkode_mem, int num_of_stages)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  switch (step_mem->LSRKmethod)
+  {
+  case ARKODE_LSRK_SSP_S_2:
+    if (num_of_stages < 2)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "num_of_stages must be greater than or equal to 2");
+      return (ARK_ILL_INPUT);
+    }
+    break;
+
+  case ARKODE_LSRK_SSP_S_3:
+    if (num_of_stages < 9 || (SUNRceil(SUNRsqrt(num_of_stages)) !=
+                              SUNRfloor(SUNRsqrt(num_of_stages))))
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__,
+                      __FILE__, "num_of_stages must be a perfect square greater than or equal to 9");
+      return (ARK_ILL_INPUT);
+    }
+    break;
+
+  case ARKODE_LSRK_SSP_10_4:
+    if (num_of_stages != 10)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "SSP10_4 method has a prefixed num_of_stages = 10");
+      return (ARK_ILL_INPUT);
+    }
+    break;
+
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Call LSRKStepSetMethod to declare SSP method type first!");
+    return (ARK_ILL_INPUT);
+    break;
+  }
+  step_mem->req_stages = num_of_stages;
```

Comment:
There should be a way to reset this to the default e.g., and input of `0` restores the default value

Reply:
I recommend that a user input `num_of_stages <= 0` indicates a reset to the default value for each method.